### PR TITLE
feat(accumulative balance): introduce scale change: by day, by week, by month

### PR DESCRIPTION
**Pull request by VladislavSokov**

### CODE REVIEW

**Comment by VladislavSokov** [comment]
Screenshot from 2023-11-11 15-15-28
Screenshot from 2023-11-11 15-15-41
Screenshot from 2023-11-11 15-16-14
Screenshot from 2023-11-11 15-16-32

**Comment by VladislavSokov** [comment]
Screenshot from 2023-11-11 16-22-25
Screenshot from 2023-11-11 16-22-16
Screenshot from 2023-11-11 16-21-55
Screenshot from 2023-11-11 16-21-43

File: db/schema.rb [hunk 1]
```diff
@@ -45,6 +45,7 @@
     t.boolean "notification", default: false
     t.datetime "archived_at"
     t.boolean "notify_parents", default: true
+    t.string "accumulative_balance_timeframe", default: "day"
```
Comment (ka8725) [inline db/schema.rb:48]: This is a user option but not account.

Let's move the persistence issue to a separate task. We should create `UserAccountSettings` model and keep this kind of data there.

**Review by ka8725 — CHANGES_REQUESTED** [review]
Great job! It works like expected, but let's move the persistence issue into a separate PR.

File: spec/models/account_spec.rb [hunk 2]
```diff
@@ -6,17 +6,60 @@
...
     let(:parent) { create(:account, :parent) }
     let(:child) { create(:account, :children, parent: parent) }
-    let!(:first_transaction) { create(:transaction, to_account: parent, from_account: child, amount: 100.0, created_at: 2.day.ago) }
-    let!(:second_transaction) { create(:transaction, to_account: child, from_account: parent, amount: 70.0, created_at: 1.day.ago) }
-    let!(:third_transaction) { create(:transaction, to_account: parent, from_account: child, amount: 50.0, created_at: Time.current) }
+    let!(:first_transaction) do
+      create(:transaction, to_account: child, from_account: parent, amount: 100.0, created_at: 1.month.ago)
+    end
+    let!(:second_transaction) do
+      create(:transaction, to_account: parent, from_account: child, amount: 70.0, created_at: 1.week.ago)
```
Comment (andreybakanovsky) [inline spec/models/account_spec.rb:13]: `let(:store) { create(:account, :store) }` -> `to_account: store`
more clear ;)

**Review by andreybakanovsky — APPROVED** [review]

**Review by ka8725 — APPROVED** [review]

**Pull request merged**